### DISCUSSION TITLE
Optimization for field magnitude (not normals) with an extended target

### DISCUSSION THREAD

**Discussion by grodgers1** [simnibs/simnibs#179] [8. Simulation workflows] [answered]
Hi SimNIBS team, thanks for the great tool!

I am trying to run an optimization for field magnitude using a target defined from a segmentation. After reading the tutorials and documentation, I could imagine two ways I could do this:
1. Use simnibs.opt_struct.TDCSDistributedOptimize(). However I do not see any option to set directions to None, and taking a peak at the code makes me think TDCSDistributedOptimize only supports field normals, not magnitudes.
2. Use simnibs.opt_struct.TDCSoptimize() and define a single target with multiple positions, i.e. a Nx3 list of positions corresponding to all voxels of the target. Then I could use target.directions=None.

Is it correct that there is no way to use option 1? Is option 2 a good alternative?

I would be curious to know how the results were achieved in G. B. Saturnino et al., Biorxiv 2020 (https://doi.org/10.1101/2020.05.27.118422), where the target in Figure 1 seems to be taken from an atlas or segmentation but the optimization was run for field magnitude.

Thanks!

**Comment by TorgeW** [comment, marked as answer]
No option 1 is currently not implemented.

Option 2 works but likely not by setting targets at all voxels. Instead you can represent your ROI by 2 or 3 positions.

**Reply by grodgers1** [reply]
Thanks, noted.
What is the rationale for taking only a few positions? I just gave it a try and using a ca. 6000x3 list of positions works well, i.e. all voxels from the segmentation. The results looked similar to using a single position plus radius of about the size of the anatomical region. Would fewer positions improve performance?
